extension/js/common/core/crypto/smime/smime-key.ts

```diff
@@ -0,0 +1,56 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+import * as forge from 'node-forge';
+import { Key, KeyUtil } from '../key.js';
+
+export class SmimeKey {
+
+  public static parse = async (text: string): Promise<Key> => {
+    const certificate = forge.pki.certificateFromPem(text);
+    const email = (certificate.subject.getField('CN') as { value: string }).value;
```

Comment:
Here would be good to try to parse the email using `Str.parseEmail` to make sure it's valid, and to provide a lowercased / normalized version of it. Else consider it an invalid pubkey

Reply:
Yes, good idea. We don't see this error normally because in the UI we check for the `emailProtection` extended key usage but a self-signed cert with this bit but malformed e-mail can exist.

I'll write a unit test for that as this (parsing) seems to be small and relatively self-contained.